tools/flaky-test-reporter/jsonreport/fakejsonreport/fakejsonreport.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakejsonreport
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"knative.dev/test-infra/tools/flaky-test-reporter/jsonreport"
+)
+
+// FakeClient fakes the jsonreport client. All file IO is redirected to data array
+type FakeClient struct {
+	data []byte
+}
+
+// Initialize wraps prow's init, which must be called before any other prow functions are used.
+func Initialize(serviceAccount string) (*FakeClient, error) {
+	return &FakeClient{}, nil
+}
+
+// CreateReport generates a flaky report for a given repository, and optionally
+// writes it to disk.
+func (c *FakeClient) CreateReport(repo string, flaky []string, writeFile bool) (*jsonreport.Report, error) {
+	report := &jsonreport.Report{
+		Repo:  repo,
+		Flaky: flaky,
+	}
+	if writeFile {
+		data, err := json.Marshal(report)
+		if err != nil {
+			return nil, err
+		}
+		c.data = data
+		return report, nil
```

Comment:
not needed. Handled by the return outside the if

Reply:
removed